lib/ansible/modules/storage/netapp/na_ontap_snmp.py

```diff
@@ -0,0 +1,152 @@
+version_added: "2.6"
+'''
+
+EXAMPLES = """
+    - name: Create SNMP community
+      na_ontap_snmp:
+        state: present
+        community_name: communityName
+        access_control: 'ro'
+        hostname: "{{ netapp_hostname }}"
+        username: "{{ netapp_username }}"
+        password: "{{ netapp_password }}"
+    - name: Delete SNMP community
+      na_ontap_snmp:
+        state: absent
+        community_name: communityName
+        access_control: 'ro'
+        hostname: "{{ netapp_hostname }}"
+        username: "{{ netapp_username }}"
+        password: "{{ netapp_password }}"
+"""
+
+RETURN = """
+"""
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+import ansible.module_utils.netapp as netapp_utils
+
+HAS_NETAPP_LIB = netapp_utils.has_netapp_lib()
+
+
+class NetAppONTAPSnmp(object):
+    '''Class with SNMP methods, doesn't support check mode'''
+
+    def __init__(self):
+
+        self.argument_spec = netapp_utils.na_ontap_host_argument_spec()
+        self.argument_spec.update(dict(
+            state=dict(required=False, type='str', choices=['present', 'absent'], default='present'),
+            community_name=dict(required=True, type='str'),
+            access_control=dict(required=True, type='str'),
+        ))
+
+        self.module = AnsibleModule(
+            argument_spec=self.argument_spec,
+            supports_check_mode=False
+        )
+
+        parameters = self.module.params
+        # set up state variables
+        self.state = parameters['state']
+        self.community_name = parameters['community_name']
+        self.access_control = parameters['access_control']
+
+        if HAS_NETAPP_LIB is False:
+            self.module.fail_json(msg="the python NetApp-Lib module is required")
+        else:
+            self.server = netapp_utils.setup_na_ontap_zapi(module=self.module)
+
+    def invoke_snmp_community(self, zapi):
+        """
+        Invoke zapi - add/delete take the same NaElement structure
+        @return: SUCCESS / FAILURE with an error_message
+        """
+        snmp_community = netapp_utils.zapi.NaElement.create_node_with_children(
+            zapi, **{'community': self.community_name,
+                     'access-control': self.access_control})
+        try:
+            self.server.invoke_successfully(snmp_community, enable_tunneling=True)
+        except netapp_utils.zapi.NaApiError:  # return False for duplicate entry
+            return False
+        return True
+
+    def add_snmp_community(self):
+        """
+        Adds a SNMP community
+        """
+        return self.invoke_snmp_community('snmp-community-add')
+
+    def delete_snmp_community(self):
+        """
+        Delete a SNMP community
+        """
+        return self.invoke_snmp_community('snmp-community-delete')
+
+    def apply(self):
+        """
+        Apply action to SNMP community
+        We want the module to be idempotent
```

Comment:
er, but it's not (at least it doesn't report as such). Is the issue that there's no way to enumerate existing communities?

Reply:
We will look at this in the next release.